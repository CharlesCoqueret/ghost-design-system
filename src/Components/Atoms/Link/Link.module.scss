.container {
  margin: auto;
  color: rgb(var(--theme-provider-charcoal));
  a,
  a:hover,
  a:focus,
  a:visited {
    color: unset;
  }

  &:hover,
  &:focus {
    color: rgb(var(--theme-provider-scooter));
  }
}

.icon {
  margin: auto 8px;
}
