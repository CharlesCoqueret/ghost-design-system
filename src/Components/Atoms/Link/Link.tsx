import React, { PropsWithChildren, ReactElement } from 'react';
import { NavLink, NavLinkProps, To } from 'react-router-dom';
import { Icon } from '../Icon';
import { Tooltip } from '../Tooltip';

import styles from './Link.module.scss';

export interface ILinkProps extends NavLinkProps {
  /** External link indicator (optional, default: false) */
  externalLink?: boolean;
  /** Url of the destination in a string or Location format */
  to: To;
  /** Tooltip content of the link (optional, default: undefined) */
  tooltip?: string;
}

const Link = (props: PropsWithChildren<ILinkProps>): ReactElement => {
  const { children, externalLink, to, tooltip, ...rest } = props;

  return (
    <div className={styles.container}>
      <NavLink
        to={to}
        {...rest}
        rel={externalLink ? 'noreferrer' : rest.rel}
        target={externalLink ? '_blank' : rest.target}>
        <Tooltip tooltip={tooltip}>
          <>{children}</>
        </Tooltip>
        {externalLink && <Icon icon={['fal', 'external-link']} className={styles.icon} size='1x' />}
      </NavLink>
    </div>
  );
};

Link.defaultProps = {
  externalLink: false,
  tooltip: undefined,
};

export default Link;
