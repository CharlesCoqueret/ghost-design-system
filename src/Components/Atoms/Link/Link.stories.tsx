import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Link, { ILinkProps } from './Link';

export default {
  title: 'Atom/Link',
  component: Link,
  parameters: { actions: { argTypesRegex: '^on.*' }, controls: { sort: 'requiredFirst' }, layout: 'centered' },
} as ComponentMeta<typeof Link>;

const Template: ComponentStory<typeof Link> = (args: ILinkProps) => {
  return <Link {...args} />;
};

export const Default = Template.bind({});
Default.args = {
  text: ' Link Text',
  link: 'https://google.com',
  tooltip: 'Information about the link in a tooltip',
};
