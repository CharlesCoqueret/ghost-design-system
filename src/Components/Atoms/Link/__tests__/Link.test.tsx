import React from 'react';
import { render } from '@testing-library/react';

import Link from '../Link';

describe('Link Component', () => {
  it('Link renders internal link', () => {
    const { container } = render(<Link to='/link'>TEXT</Link>);
    expect(container).toMatchSnapshot();
  });

  it('Link renders external link with tooltip', () => {
    const { container } = render(
      <Link to='http://www.link.com' tooltip='TOOLTIP' externalLink>
        TEXT
      </Link>,
    );
    expect(container).toMatchSnapshot();
  });
});
