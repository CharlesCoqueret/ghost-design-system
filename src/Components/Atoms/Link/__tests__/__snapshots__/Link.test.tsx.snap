// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component Link renders external link with tooltip 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      partial url
      _blank
      TEXT
      <div>
        TOOLTIP
      </div>
      <i
        class="icon icon"
        icon="fal,external-link"
      />
    </div>
  </div>
</div>
`;

exports[`Link Component Link renders internal link 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      partial url
      TEXT
    </div>
  </div>
</div>
`;
